boost-maven/boost-maven-plugin/src/main/java/io/openliberty/boost/maven/plugin/AbstractMojo.java

```diff
@@ -88,31 +99,34 @@ public void execute() throws MojoExecutionException {
     }
     
     protected RuntimeI getRuntimeInstance() throws MojoExecutionException {
-        if(runtime == null) {
+        if (runtime == null) {
             BoostLogger logger = BoostLogger.getInstance();
-            if (dependencies.containsKey(AbstractBoosterConfig.RUNTIMES_GROUP_ID + ":tomee")) {
-                logger.info("Detected TomEE as target Boost runtime");
-                String tomeeInstallDir = projectBuildDir + "/apache-tomee/";
-                String tomeeConfigDir = tomeeInstallDir + "conf";
-                runtime = new TomeeRuntime(dependencies, getExecutionEnvironment(), tomeeInstallDir, tomeeConfigDir, getMavenDependencyPlugin());
-            } else if (dependencies.containsKey(AbstractBoosterConfig.RUNTIMES_GROUP_ID + ":openliberty")) {
-                logger.info("Detected Open Liberty as target Boost runtime");
-                String runtimeGroupId = "io.openliberty";
-                String runtimeArtifactId = "openliberty-runtime";
-                String runtimeVersion = "19.0.0.3";
-                runtime = new LibertyRuntime(dependencies, getExecutionEnvironment(), project, getLog(), repoSystem, repoSession, remoteRepos, getMavenDependencyPlugin(), runtimeGroupId, runtimeArtifactId, runtimeVersion);
-            } else if (dependencies.containsKey(AbstractBoosterConfig.RUNTIMES_GROUP_ID + ":wlp")) {
-                logger.info("Detected WebSphere Liberty as target Boost runtime");
-                String runtimeGroupId = "com.ibm.websphere.appserver.runtime";
-                String runtimeArtifactId = "wlp-javaee7";
-                String runtimeVersion = "19.0.0.3";
-                runtime = new LibertyRuntime(dependencies, getExecutionEnvironment(), project, getLog(), repoSystem, repoSession, remoteRepos, getMavenDependencyPlugin(), runtimeGroupId, runtimeArtifactId, runtimeVersion);
-            }
-            else {
-                throw new MojoExecutionException("No target Boost runtime was detected. Please add a runtime and restart the build.");
+            RuntimeParams params = new RuntimeParams(dependencies, getExecutionEnvironment(), project, getLog(), repoSystem, repoSession, remoteRepos, getMavenDependencyPlugin());
+            try {
+                List<URL> pathUrls = new ArrayList<URL>();
+                for(String compilePathElement : project.getCompileClasspathElements()) {
+                    pathUrls.add(new File(compilePathElement).toURI().toURL());
+                }
+
+                URL[] urlsForClassLoader = pathUrls.toArray(new URL[pathUrls.size()]);
+                ClassLoader compileClassLoader = new URLClassLoader(urlsForClassLoader, this.getClass().getClassLoader());
+                List<Class<?>> runtimes = ClassFinderUtil.findClassesImplementing(RuntimeI.class, "boost.runtimes", compileClassLoader, logger);
```

Comment:
Hmm... I haven't thought this through extenisvely, but would've expected maybe something like a ServiceLoader extension...e.g. an `src/main/resources/META-INF/services/io.microprofile.boost.RuntimeI` with the contents equal to an impl class name.      It seems this assumes a convention that all impls use the **boost.runtimes** package (or package prefix maybe?  didn't digest code fully).   Is there a precedent for that sort of extension mechanism over something like **ServiceLoader**?

Reply:
Ya, didn't know about this approach. I'll take a look.

Reply:
The path would match the intf qualified class name, so for the code as-is it would actually be `src/main/resources/META-INF/services/io.openliberty.boost.common.runtimes.RuntimeI`.